module/VuFind/src/VuFind/View/Helper/Root/SortFacetList.php

```diff
@@ -67,9 +69,27 @@ public function __invoke($results, $field, $list, $searchRoute)
             $url = $urlHelper($searchRoute) . $results->getUrlQuery()
                 ->addFacet($field, $value['value'], ($value['operator'] ?? 'AND'))
                 ->getParams();
-            $facets[$url] = $value['displayText'];
+            $facets[$url] = $this->facetValueToString($value, $formatString);
         }
         $this->getSorter()->natsort($facets);
         return $facets;
     }
+
+    /**
+     * Convert a facet array to a string.
+     *
+     * @param array  $value  Facet value array
+     * @param string $format Format string
+     *
+     * @return string
+     */
+    protected function facetValueToString(array $value, string $format): string
+    {
+        $search = $replace = [];
+        foreach ($value as $key => $value) {
```

Comment:
This does not result in an error but it would probably be safer to avoid the duplication of `$value`.

Reply:
Good catch! Fixed in [4d0ad51](https://github.com/vufind-org/vufind/pull/3621/commits/4d0ad5172048f9491f67a3c5a0a08ec5d8bcc630).